client/src/components/RenderFile/RenderFile.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+import useStyles from './useStyles';
+import { Box, Button, Modal, Typography } from '@material-ui/core';
+import { uploadProfileImage } from '../../helpers/APICalls/upload';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { useAuth } from '../../context/useAuthContext';
+
+interface Props {
+  file: any;
+  onHandleClose: () => void;
+  open: boolean;
+  onSetFile: any;
+  setOpen: any;
+}
+
+const RenderFile = ({ file, onHandleClose, open, onSetFile, setOpen }: Props): JSX.Element => {
+  const sizeInBytes = `${(parseInt(file?.size) / (1024 * 1024)).toFixed(2)} MB`;
+  const fileFormat = file?.type.split('/')[1];
+  const classes = useStyles();
+  const [uploadState, setUploadState] = useState('');
+  const { updateSnackBarMessage } = useSnackBar();
+  const { updateLoginContext } = useAuth();
+
+  const handleUpload = () => {
+    if (uploadState === 'uploading...') {
+      return;
+    }
+    setUploadState('uploading...');
+    uploadProfileImage(file).then((data) => {
+      if (data.error) {
+        updateSnackBarMessage(data.error.message);
+        setUploadState('picture upload failed!');
+        onSetFile(null);
+        setOpen(false);
+        setTimeout(() => {
+          setUploadState('');
+        }, 3000);
```

Comment:
How could we are timing out here?

Reply:
Ideally, it should be fast and responsive. If the users network is slower they will see the loading state longer. I don't think we need to unnecessarily show a uploading state if their network is fast

Reply:
ok, if not for showing the upload state, the upload state is still used to prevent the user from sending request whiles the ongoing one is not done
![2021-11-13_121920](https://user-images.githubusercontent.com/70457144/141643571-fa2f490e-92f3-401b-9139-f5f374ea838e.png)